src/DataBase/UserDatabase.cs

```diff
@@ -43,36 +51,60 @@ public static int Authenticate(string email, string pass)
             return user.Id;
         }
 
-        public static bool ChangePassword(string email, string pass)
+        public async Task<bool> ChangePasswordAsync(string email, string pass)
         {
-            using var db = new DatabaseContext();
-            var user = db.Users.FirstOrDefault(a => a.Email == email);
+            var user = await Database.Users.FirstOrDefaultAsync(a => a.Email == email);
 
             if (user == null)
             {
                 return false;
             }
-                
+
             var salt = HashingProcessor.CreateSalt();
             var passwordHash = HashingProcessor.GenerateHash(pass, salt);
 
             user.Password = passwordHash;
             user.Salt = salt;
-            db.SaveChanges();
+            await Database.SaveChangesAsync();
+
+            return true;
+        }
+
+        public async Task<bool> DeleteUserAsync(int userId)
+        {
+            return await DeleteUserAsync(x => x.Id == userId);
+        }
 
+        public async Task<bool> DeleteUserAsync(string email)
+        {
+            return await DeleteUserAsync(x => x.Email == email);
+        }
+
+        public async Task<bool> DeleteUserAsync(Expression<Func<UserModel, bool>> filter)
+        {
+            var temp = await Database.Users.FirstOrDefaultAsync(filter);
+            if (temp is null)
+            {
+                return false;
+            }
+            Database.Users.Remove(temp);
+            await Database.SaveChangesAsync();
             return true;
         }
 
-        public static UserModel GetUser(int userId)
+        public async Task<UserModel> GetUserAsync(int userId)
+        {
+            return await GetUserAsync(x => x.Id == userId);
+        }
+
```

Comment:
What piece of info this returns?

Reply:
This returns UserModel ( I wanted to make this as reusable as possible, for example instead of having two different methods to get user id or user email, you just get this user model object and you can extract all the info related from the user from it)